src/transformState.js

```diff
@@ -5,7 +5,28 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete state[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        return 'invalid type of action';
```

Comment:
it's just string, check prev comments
![image](https://user-images.githubusercontent.com/59209337/206132728-96f59e8d-d716-40f6-bb6d-b434d72155d2.png)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error